tests/aws/services/sts/test_sts.py

```diff
@@ -92,8 +93,53 @@
 
 
 class TestSTSIntegrations:
-    @markers.aws.unknown
-    def test_assume_role(self, aws_client):
+    @markers.aws.validated
+    @markers.snapshot.skip_snapshot_verify(
+        paths=["$..PackedPolicySize"],
+    )
+    def test_assume_role(self, aws_client, create_role, account_id, snapshot):
+        snapshot.add_transformers_list(
+            [
+                snapshot.transform.resource_name(),
+                snapshot.transform.key_value("RoleId"),
+                snapshot.transform.key_value("AccessKeyId"),
+                snapshot.transform.key_value("SecretAccessKey"),
+                snapshot.transform.key_value("SessionToken"),
+            ]
+        )
+
+        test_role_session_name = f"test-assume-role-{short_uid()}"
+        test_role_name = f"role-{short_uid()}"
+        assume_policy_doc = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Action": "sts:AssumeRole",
+                    "Principal": {"AWS": account_id},
+                    "Effect": "Allow",
+                }
+            ],
+        }
+        created_role = create_role(
+            RoleName=test_role_name, AssumeRolePolicyDocument=json.dumps(assume_policy_doc)
+        )
+        snapshot.match("create-role", created_role)
+
+        def assume_role():
+            assume_role_resp = aws_client.sts.assume_role(
+                RoleArn=created_role["Role"]["Arn"], RoleSessionName=test_role_session_name
+            )
+            return assume_role_resp
+
+        response = retry(assume_role, sleep=5, retries=4)
+        snapshot.match("assume-role", response)
+
+        if response["AssumedRoleUser"]["AssumedRoleId"]:
+            assume_role_id_parts = response["AssumedRoleUser"]["AssumedRoleId"].split(":")
+            assert assume_role_id_parts[1] == test_role_session_name
```

Comment:
Isn't that a redundant check, since we already have a snapshot for the response? Aws seems to use the `RoleSessionName` in the ARN, so it is validating that both part are equal already?

```json
        "AssumedRoleUser": {
          "Arn": "arn:aws:sts::111111111111:assumed-role/<resource:1>/<resource:2>",
          "AssumedRoleId": "<role-id:1>:<resource:2>"
        },
```

Reply:
Yes, I think the snapshot validates that the second part of `AssumeRoleId` is the same as the last part of the `Arn` resource.

But from the snapshot, nothing tells us that the last part of the `AssumedRoleUser` is the `test_role_session_name` I would say. It's kinda implicit looking at it? I could also instead of this check, just create a snapshot entry `snapshot.match("role-session-name", {"RoleSessionName": test_role_session_name})` and then create a transformer on it, that might clearer than `"<resource:2>"`?

Reply:
Updated with https://github.com/localstack/localstack/pull/10962/commits/475a6f600e2c5ece235dd609cbbaac4128009a65, tell me if that's clearer 😄

Reply:
Love it! :heart: